pkg/svc/workflow.go

```diff
@@ -161,12 +207,37 @@ func newRestWorkflowAssignment(workflowName string, codesets []*domain.Codeset,
 	}
 	for _, c := range codesets {
 		assignment.Codesets = append(assignment.Codesets, &workflow.Codeset{
-			Name:        c.Name,
-			Project:     c.Project,
-			Description: c.Description,
-			Labels:      c.Labels,
-			URL:         &c.URL,
+			Name:        c.Codeset.Name,
+			Project:     c.Codeset.Project,
+			Description: c.Codeset.Description,
+			Labels:      c.Codeset.Labels,
+			URL:         &c.Codeset.URL,
 		})
 	}
 	return assignment
 }
+
+func (s *workflowsrvc) unassignCodesetFromWorkflow(ctx context.Context, workflowName string, codeset *domain.Codeset) (err error) {
+	assignment := s.store.GetAssignedCodeset(ctx, workflowName, codeset)
+	if assignment == nil {
+		err = fmt.Errorf("workflow not assigned to codeset")
+		s.logger.Print(err)
+		return workflow.MakeNotFound(err)
+	}
+
+	err = s.codesetStore.DeleteWebhook(ctx, codeset, assignment.WebhookID)
+	if err != nil {
+		s.logger.Print(err)
+		return err
+	}
+
+	if len(s.store.GetAssignedCodesets(ctx, workflowName)) == 1 {
```

Comment:
Just for my clarification (would be nice to document this as a comment though): you only delete the listener when the last codeset is un-assigned from the workflow, is that right ?